graphql-dgs-codegen-core/src/main/kotlin/com/netflix/graphql/dgs/codegen/generators/java/DataTypeGenerator.kt

```diff
@@ -522,6 +545,17 @@ abstract class BaseDataTypeGenerator(
         javaType.addField(fieldBuilder.build())
         javaType.addMethod(getterMethodBuilder.build())
         javaType.addMethod(setterMethodBuilder.build())
+        if (fieldDefinition.trackFieldSet) {
+            val hasFieldMethodName = typeUtils.transformIfDefaultClassMethodExists("has${fieldDefinition.name[0].uppercase()}${fieldDefinition.name.substring(1)}", TypeUtils.setClass)
+            val hasFieldMethodBuilder = MethodSpec.methodBuilder(hasFieldMethodName)
```

Comment:
I'm wondering if we should expose the `Optional` instead of proving a boolean `has*` method. E.g. we could add a `getOptional[FieldName]` method for each field besides the `get[FieldName]` method. WDYT?